tasks/PIDController.cpp

```diff
@@ -44,21 +44,18 @@ bool PIDController::startHook()
 void PIDController::updateHook()
 {
     RTT::FlowStatus status = _pose_samples.readNewest(pose_sample);
-    if (status == RTT::NoData){
-        if(state() != WAIT_FOR_POSE_SAMPLE){
-            error(WAIT_FOR_POSE_SAMPLE);
-        }
-        return;
-    }
-    else if (status == RTT::OldData && new_pose_samples_timeout.elapsed()){
-        if(state() != WAIT_FOR_POSE_SAMPLE){
-            error(WAIT_FOR_POSE_SAMPLE);
+    if (status != RTT::NewData)
+    {
+        if(new_pose_samples_timeout.elapsed())
+        {
+            exception(POSE_TIMEOUT);
```

Comment:
I'm fine with separating the states to a soft and a hard reference sample timeout.
But we should adapt the other components as well in order to have an equal behavior.